### PR TITLE
Interpreter store instruction fix: issue 148

### PR DESCRIPTION
**Pull request by zzzDavid**
### Description

This pull request is to solve issue: https://github.com/sampsyo/bril/issues/148.

### Changes
Changed the get argument switch case in `store` instruction to `getArgument` function to support any-type store.

### CODE REVIEW

File: bril-ts/brili.ts [hunk 1]
```diff
@@ -629,20 +629,7 @@ function evalInstr(instr: bril.Instruction, state: State): Action {
...
-      case "bool": {
-        state.heap.write(target.loc, getBool(instr, state.env, 1));
-        break;
-      }
-      default: {
-        state.heap.write(target.loc, getPtr(instr, state.env, 1));
-        break;
-      }
-    }
+    state.heap.write(target.loc, getArgument(instr, state.env, 1));
```
Comment (sampsyo) [inline bril-ts/brili.ts:632]: Consider passing `target.type` as the fourth argument to `getArgument` to enable type checking.

**Review by sampsyo — CHANGES_REQUESTED** [review]
Thanks for getting this started so quickly!!

It looks like the test `wrong_write.bril` is failing because we no longer throw an error when trying to store a `bool` value into a `ptr<int>` pointer. Adding some type checking back to the `getArgument` call would help.

It would also be extra-awesome to add your short test program from #148 as a new test so we can be sure that `ptr<float>` is now working in the interpreter.

File: bril-ts/brili.ts (hunk 1 above)
Comment (zzzDavid) [reply bril-ts/brili.ts]: Thanks! I just added that, it reminds me always check types before writing ;)

**Comment by sampsyo** [comment]
Cool—what do you think about the above suggestion to include your test case?

**Comment by zzzDavid** [comment]
> Cool—what do you think about the above suggestion to include your test case?

Thanks, just added that

**Review by sampsyo — APPROVED** [review]
Looks perfect; thanks!

**Pull request merged**